docker.go

```diff
@@ -675,6 +675,7 @@ func (p *DockerProvider) CreateContainer(ctx context.Context, req ContainerReque
 		Labels:       req.Labels,
 		Cmd:          req.Cmd,
 		Hostname:     req.Hostname,
+		User:         req.User,
```

Comment:
What would happen if the `req.User` is empty? Will the docker client understand that as `root`?

Reply:
`User` is a string field, so it appears that an empty string equates to not setting a UID at all:

Confirmed with these tests:

```go
package main

import (
	"bytes"
	"context"
	"fmt"

	"github.com/testcontainers/testcontainers-go"
	"github.com/testcontainers/testcontainers-go/wait"
)

func main() {
	tests := []testcontainers.ContainerRequest{
		{
			Image:      "debian:latest",
			Cmd:        []string{"sh", "-c", "id -u"},
			WaitingFor: wait.ForExit(),
		},
		{
			Image:      "debian:latest",
			User:       "60125",
			Cmd:        []string{"sh", "-c", "id -u"},
			WaitingFor: wait.ForExit(),
		},
	}

	for _, test := range tests {
		ctx := context.Background()
		c, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
			ContainerRequest: test,
			Started:          true,
		})
		if err != nil {
			panic(err)
		}
		buf := new(bytes.Buffer)
		r, _ := c.Logs(ctx)
		buf.ReadFrom(r)
		fmt.Printf("User: expected: %s; got: %s\n", test.User, buf.String())
	}
}

```

Output:

```
2021/10/05 08:17:34 Starting container id: fa62829446d6 image: quay.io/testcontainers/ryuk:0.2.3
2021/10/05 08:17:35 Waiting for container id fa62829446d6 image: quay.io/testcontainers/ryuk:0.2.3
2021/10/05 08:17:35 Container is ready id: fa62829446d6 image: quay.io/testcontainers/ryuk:0.2.3
2021/10/05 08:17:35 Starting container id: 1b75176f40da image: debian:latest
2021/10/05 08:17:35 Waiting for container id 1b75176f40da image: debian:latest
2021/10/05 08:17:35 Container is ready id: 1b75176f40da image: debian:latest
User: expected: ; got: 0

2021/10/05 08:17:35 Starting container id: 063145d38a97 image: debian:latest
2021/10/05 08:17:36 Waiting for container id 063145d38a97 image: debian:latest
2021/10/05 08:17:36 Container is ready id: 063145d38a97 image: debian:latest
User: expected: 60125; got: 60125
```

Reply:
Would you mind adding two tests in your PR: one for adding the User, and another one without it?

Besides that, we could only set the `User` field if `req.User` is not empty

Reply:
@mdelapenya sure thing!